### PR TITLE
Add example for `newtype` parameter in routes

### PR DESCRIPTION
**Pull request by brodeuralexis**
Also add parens to previous examples that failed compilation.

### CODE REVIEW

File: GUIDE.md [hunk 1]
```diff
@@ -336,3 +336,36 @@ main = do
     Right { foo, bar } -> ...
     Left errors -> ...
 ```
+
+## Using `newtype` in routes
+
+Using the following routes with a `newtype`:
+```purescript
+newtype PostId = PostID Int
+derive instance newtypePostID :: Newtype PostId _
```
Comment (natefaubion) [inline GUIDE.md]: Some inconsistent casing (PostId vs PostID).

File: GUIDE.md [hunk 2]
```diff
@@ -336,3 +336,36 @@ main = do
...
+
+data MyRoute
+  = PostIndex
+  | Post PostId
+  | PostEdit PostId
+  | PostBrowse String String
+```
+
+It is possible to warp an `int` route parameter into a `PostId` using the
```
Comment (natefaubion) [inline GUIDE.md]: Typo "warp"

File: GUIDE.md [hunk 3]
```diff
@@ -336,3 +336,36 @@ main = do
...
+following function:
+```purescript
+postId :: forall f. MatchClass f => f PostId
+postId = pure wrap <*> int
```
Comment (natefaubion) [inline GUIDE.md]: This is `wrap <$> int`, but I would prefer `PostId <$> int`.

File: GUIDE.md [hunk 4]
```diff
@@ -336,3 +336,36 @@ main = do
...
+```
+
+The created `postId` function can then be used like the `str` function.
```
Comment (natefaubion) [inline GUIDE.md]: `str` isn't a function. I would say "parser" instead.

**Comment by JordanMartinez** [comment]
I've addressed Nate's feedback above, but we still need a changelog entry file, which can't be added in this PR AFAIK unless the original author adds that change.

**Review by JordanMartinez — APPROVED** [review]

**Review by thomashoneyman — APPROVED** [review]

**Pull request merged**